Mathlib/Topology/Algebra/Module/ModuleTopology.lean

```diff
@@ -304,4 +315,85 @@ theorem continuous_of_ringHom {R A B} [CommSemiring R] [Semiring A] [Algebra R A
 
 end function
 
+section surjection
+
+variable {R : Type*} [τR : TopologicalSpace R] [Ring R]
+variable {A : Type*} [AddCommGroup A] [Module R A] [TopologicalSpace A] [IsModuleTopology R A]
+variable {B : Type*} [AddCommGroup B] [Module R B] [τB : TopologicalSpace B] [IsModuleTopology R B]
+
+open Topology in
+/-- A linear surjection between modules with the module topology is a quotient map.
+Equivalently, the pushforward of the module topology along a surjective linear map is
+again the module topology. -/
```

Comment:
But it isn't Lean-straightforward to deduce that from this statement, is it? Because you can only apply this statement if you already have an `IsModuleTopology R B`.
Do you think you can factor out this pushforward claim into a prerequisite lemma?

Reply:
I've factored it out into a corollary. In fact you've spotted a glitch in the PR (now fixed): in an earlier version of this work I was claiming as the main (only) result that the pushforward of the module topology was the module topology, and then after discussions with Yury he convinced me that it was much cleaner to state that a surjection was a quotient map, so I restated the result to the thing you now see, but forgot to change the name. I've now changed the name of the main lemma to  `IsModuleTopology.isQuotientMap_of_surjective` (which is what I prove) and have added the corollary `ModuleTopology.eq_coinduced_of_surjective` which says what it's supposed to say. In applications I expect `IsModuleTopology` to be more widely used than `moduleTopology` so I am trying to frame everything in this language.

Reply:
Great! Finally, do you want to mention this result under "Main theorems"?

bors d+